tests/unit/file_io_test.cc

```diff
@@ -947,3 +950,20 @@ SEASTAR_TEST_CASE(test_oversized_io_works) {
         BOOST_REQUIRE((size_t)std::count_if(buf.get(), buf.get() + buf_size, [](auto x) { return x == 'a'; }) == buf_size);
     });
 }
+
+SEASTAR_TEST_CASE(test_file_system_space) {
+    return tmp_dir::do_with_thread([] (tmp_dir& t) {
+        const auto& name = t.get_path().native();
+        struct statvfs st;
+        space_info si, si1;
+        do {
+            si = file_system_space(name).get();
+            st = engine().statvfs(name).get();
+            si1 = file_system_space(name).get();
+        } while (si1 != si);
+
+        BOOST_REQUIRE_EQUAL(st.f_blocks * st.f_frsize, si.capacity);
+        BOOST_REQUIRE_EQUAL(st.f_bfree * st.f_frsize, si.free_space);
+        BOOST_REQUIRE_EQUAL(st.f_bavail * st.f_frsize, si.available_space);
```

Comment:
This isn't guaranteed to pass. If a byte is added to the filesystem after assignment to si and removed after assignment to st, si == si1 && si != st.

There isn't a simple way to test this.

Reply:
Without syncing the filesystem, it is auto-synced rarely (every 30 seconds?), so we shouldn't see these fluctuations in practice.

Reply:
Who says space is updated on sync?

Reply:
Good point, it's probably file system specific.
I saw it with xfs, but tmpfs indeed updates it synchronously.